### ISSUE TITLE
[v2.08] Timer Failed! - Dusk/dawn events

### ISSUE THREAD

**Issue by Peekofwar** [Nak2/StormFox2#8]
I've noticed that whenever the game time or weather ticks over, the dusk and dawn events don't seem to trigger. I have two relays in my map called `dusk` and `dawn`. 

This particular error happened at dawn:
```
[StormFox 2] lua/stormfox2/functions/sv_mapio.lua:22: Tried to use a NULL entity!
  1. Fire - [C]:-1
   2. setELight - lua/stormfox2/functions/sv_mapio.lua:22
    3. unknown - lua/stormfox2/functions/sv_mapio.lua:31

Timer Failed! [StormFox2.mi.lights][@lua/stormfox2/functions/sv_mapio.lua (line 29)]
```

1. Set the time a few minutes prior to either dusk or dawn.
2. Allow Stormfox to tick over the time.
3. There should be a script error in the console.
4. Relays won't have triggered.

![image](https://user-images.githubusercontent.com/10189526/115102230-664aec00-9efe-11eb-822d-28ada976f222.png)
![image](https://user-images.githubusercontent.com/10189526/115102213-46b3c380-9efe-11eb-8672-cb9f22f02bf4.png)
![20210417005142_1](https://user-images.githubusercontent.com/10189526/115102137-f50b3900-9efd-11eb-9f04-ea8ffefb4708.jpg)

**Comment by Nak2** [comment]
SF2 will try and turn on/off all "env_projectedtexture", "light_dynamic", "light" and "light_spot"'s on the ma,p if they meet the following criterias:
- If the entity has "night" or "day" as part of their targetname or named "1". ( or no targetname for light_spot )
- The entitys targetname doesn't have "indoor" in the name.

The error happens because the entity it tried to toggle got deleted. I've made a patch.
Would it be possible for you to upload your map so I can test it?

**Comment by Peekofwar** [comment]
> Would it be possible for you to upload your map so I can test it?

https://steamcommunity.com/sharedfiles/filedetails/?id=2365722107

**Comment by Peekofwar** [comment]
Heads-up: The next update to the map moves some stuff into instances including the lanterns. This means that the lantern's relays will change from `dawn` to `sl1_dawn` for example.

**Comment by Nak2** [comment]
Should be fixed in 2.09.
Problem seem to be that weather on the server was nil, therefore never sets the light or update the logic_relays.

Feel free to reopen if it still have problems.
